File: LinkedList/typescript/linked_list_cycle.ts
/**
 * Definition for singly-linked list.
*/
 class ListNode {
    val: number
    next: ListNode | null
    constructor(val?: number, next?: ListNode | null) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

function hasCycle(head: ListNode | null): boolean {
    let slow = head
    let first = head
    let flag:boolean = false
    while(first!=null && first.next!=null && slow!=null){
        slow = slow.next
        first = first.next.next
        if(slow===first){
            flag = true
            break
        }
    }
    return flag
};

File: LinkedList/python/linked_list_cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        slow = head
        first = head
        flag =False
        while first != None and first.next != None:
            slow = slow.next
            first = first.next.next
            if slow == first:
                flag =True
                break
        return flag
